### PR TITLE
Proxy 9 - 13

### PR DESCRIPTION
**Pull request by MariKov**

### CODE REVIEW

File: app/delivery/controllers/mars_rover_controller.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+const getMarsRoverPhoto = require("../../use_cases/mars_rover/mars_rover_service");
+
+async function handlePostForMarsRover(req, res) {
+    try {
+        console.log(req.body.userId, req.body.userName);
```
Comment (mdziarkach) [inline app/delivery/controllers/mars_rover_controller.js]: probably you forgot to remove it

File: app/delivery/controllers/mars_rover_controller.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+        const marsPhotoImageUrl = await getMarsRoverPhoto(req.body.apiKey);
+        res.render('mars_rover/mars_rover_photo.njk', {marsPhotoImageUrl});
+    } catch (error) {
+        res.status(500).send("Ups, something went wrong: " + error.message);
```
Comment (mdziarkach) [inline app/delivery/controllers/mars_rover_controller.js]:
```js
next(error)
```
don't forget to add error handler as a last middleware if it doesn't exists

File: app/delivery/middlewares/validator.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+const validateQueryParams = (schema) => (req, res, next) => {
+    const {error} = schema.validate(req.query);
+    if (error) {
+        res.status(400).json({error: error.details[0].message});
```
Comment (mdziarkach) [inline app/delivery/middlewares/validator.js]: you can throw exception here

File: app/delivery/middlewares/validator.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    } else {
```
Comment (mdziarkach) [inline app/delivery/middlewares/validator.js]: else is not required here
```js
if (error) {...}
next()
```

File: app/delivery/middlewares/validator.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+        next();
+    }
+};
+
+const validateRequestBody = (schema) => (req, res, next) => {
+    const {error} = schema.validate(req.body);
+    if (error) {
+        res.status(400).json({error: error.details[0].message});
```
Comment (mdziarkach) [inline app/delivery/middlewares/validator.js]: please throw custom exception here

File: app/delivery/views/meteors/meteors_view.njk [hunk 6]
```diff
@@ -0,0 +1,29 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <title> Meteors </title>
+</head>
+<body>
```
Comment (mdziarkach) [inline app/delivery/views/meteors/meteors_view.njk]: you can create layout.njk template and reuse it in several places

File: app/use_cases/meteors/meteors_service.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
+const mapAsteroids = require("./meteors_transformer");
+const {fetchAsteroids} = require("../../repository/nasa_api");
+
+async function getMeteors(startDate, endDate, isDangerous, count) {
+    try {
+        const asteroids = await fetchAsteroids(startDate, endDate);
+        return mapAsteroids(asteroids, filterPredicate(isDangerous), count);
+    } catch (error) {
+        console.error('Ups, something went wrong: ', error);
+        throw error;
```
Comment (mdziarkach) [inline app/use_cases/meteors/meteors_service.js]: create please custom Exception with possibility to provide statusCode and errorMessage

File: app/utils/formatters.js [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+function isDate(input_date) {
+    return input_date && moment(input_date).isValid();
+}
+
+function formatDate(input_date) {
+    return input_date ? moment(input_date).format('YYYY-MM-DD') : input_date;
+}
+
+function formatBoolean(input) {
+    return typeof input === "boolean"
```
Comment (mdziarkach) [inline app/utils/formatters.js]: is it possible to make it easier?

File: app/delivery/controllers/mars_rover_controller.js (hunk 1 above)
Comment (MariKov) [reply app/delivery/controllers/mars_rover_controller.js]: Yes😎 Deleted now

File: app/delivery/controllers/mars_rover_controller.js (hunk 2 above)
Comment (MariKov) [reply app/delivery/controllers/mars_rover_controller.js]: Done!

File: app/delivery/middlewares/validator.js (hunk 3 above)
Comment (MariKov) [reply app/delivery/middlewares/validator.js]: Done!

File: app/delivery/middlewares/validator.js (hunk 5 above)
Comment (MariKov) [reply app/delivery/middlewares/validator.js]: Done!

File: app/use_cases/meteors/meteors_service.js (hunk 7 above)
Comment (MariKov) [reply app/use_cases/meteors/meteors_service.js]: Added error handling, thank you for paying my attention to it☺

File: app/utils/formatters.js (hunk 8 above)
Comment (MariKov) [reply app/utils/formatters.js]: Tryied another approach, really looks better, is it better?)

File: app/delivery/middlewares/validator.js (hunk 4 above)
Comment (MariKov) [reply app/delivery/middlewares/validator.js]: Done!

File: app/delivery/views/meteors/meteors_view.njk (hunk 6 above)
Comment (MariKov) [reply app/delivery/views/meteors/meteors_view.njk]: Updated

**Review by mdziarkach — APPROVED** [review]
now it looks good to me

**Pull request merged**